flink-connectors/flink-connector-pubsub/src/main/java/org/apache/flink/streaming/connectors/pubsub/Bound.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.pubsub;
+
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * This class defines a bound based on messages received or time since last received message.
+ * Using start(SourceFunction) starts the bound. Everytime a message is received the sourceFunction should call receivedMessage().
+ * When the bound is reached, the sourcefunction gets closed by calling sourceFunction.close()
+ * See {@link BoundedPubSubSource}.
+ *
+ * @param <OUT> type of message that is received by the SourceFunction.
+ */
+class Bound<OUT> implements Serializable {
+	private static final Logger LOG = LoggerFactory.getLogger(Bound.class);
+
+	private final Bound.Mode mode;
+	private final long maxMessagedReceived;
+	private final long maxTimeBetweenMessages;
+
+	private SourceFunction<OUT> sourceFunction;
+	private transient Timer timer;
+	private long messagesReceived;
+	private long lastReceivedMessage;
+	private boolean cancelled = false;
+
+	private Bound(Bound.Mode mode, long maxMessagedReceived, long maxTimeBetweenMessages) {
+		this.mode = mode;
+		this.maxMessagedReceived = maxMessagedReceived;
+		this.maxTimeBetweenMessages = maxTimeBetweenMessages;
+		this.messagesReceived = 0L;
```

Comment:
The cancelled variable is set in the class, and the messageReceived in the constructor, maybe choose either one of them? To keep the code in the same style.